### PR TITLE
feat: CARITAS-297-add-site24x7

### PR DESCRIPTION
**Pull request by Leandro13Silva13**
add configs for site24x7

Fixes #

## Proposed Changes

  -
  -
  -

### CODE REVIEW

**Comment by vi-sudo** [comment]
Task linked: [CARITAS-297 [Caritas] POC: site24x7](https://app.clickup.com/t/6656196/CARITAS-297)

File: templates/userservice/userservice-deployment.yaml [hunk 1]
```diff
@@ -491,6 +491,18 @@ spec:
...
+              valueFrom:
+                secretKeyRef:
+                  key: S247_LICENSE_KEY
+                  name: userservice-secret-{{ .Release.Name }}-env
+            - name: JAVA_TOOL_OPTIONS
+              valueFrom:
+                configMapKeyRef:
+                  key: JAVA_TOOL_OPTIONS
+                  name: userservice-{{ .Release.Name }}-env
+          { { - end } }
```
Comment (tkuzynow) [inline templates/userservice/userservice-deployment.yaml:505]: same issue with brackets here

File: templates/userservice/userservice-env-configmap.yaml [hunk 2]
```diff
@@ -105,6 +105,10 @@ data:
   {{- end }}
   SPRING_SLEUTH_WEB_ENABLED: {{ .Values.zipkin.enabled | default false | quote }}
   SPRING_SLEUTH_SAMPLER_PERCENTAGE: {{ .Values.zipkin.samplerPercentage | default "1.0" | quote }}
+  # 24x7
```
Comment (tkuzynow) [inline templates/userservice/userservice-env-configmap.yaml:108]: not sure if this is proper comment, i'd remove it

File: values.yaml [hunk 3]
```diff
@@ -14,6 +14,7 @@ global:
   useExternalMariaDb: false
   settingMainTenantSubdomainForSingleDomainMultitenancy: "not-needed"
   migrationToolEnabled: false
+  24x7Enabled: false
```
Comment (tkuzynow) [inline values.yaml:17]: could you change variable name to site24x7Enabled?

File: values.yaml [hunk 4]
```diff
@@ -310,4 +311,7 @@ weblate:
 migration:
   weblate:
     useCustomResourcesPath: true
-    customResourcesPath: "resources/weblate/"
\ No newline at end of file
+    customResourcesPath: "resources/weblate/"
+
+# site24x7
+s247licenseKey: "key"
```
Comment (tkuzynow) [inline values.yaml:317]: change variable name from s247... to site24x7LicenseKey

**Review by tkuzynow — APPROVED** [review]

File: templates/userservice/userservice-env-configmap.yaml [hunk 5]
```diff
@@ -105,6 +105,9 @@ data:
   {{- end }}
   SPRING_SLEUTH_WEB_ENABLED: {{ .Values.zipkin.enabled | default false | quote }}
   SPRING_SLEUTH_SAMPLER_PERCENTAGE: {{ .Values.zipkin.samplerPercentage | default "1.0" | quote }}
+  {{ - if .Values.global.site24x7Enabled }}
+  JAVA_TOOL_OPTIONS: "-javaagent:/opt/apm/apminsight-javaagent.jar -Dapminsight.application.name=userservice-{{ .Release.Name }} -Dapminsight.log.dir=/home/apm"
```
Comment (tkuzynow) [inline templates/userservice/userservice-env-configmap.yaml:109]: you could parametrize name userservice, but it's also fine like this

File: templates/userservice/userservice-deployment.yaml [hunk 6]
```diff
@@ -491,6 +491,18 @@ spec:
                 configMapKeyRef:
                   key: SPRING_SLEUTH_SAMPLER_PERCENTAGE
                   name: userservice-{{ .Release.Name }}-env
+          {{ - if .Values.global.site24x7Enabled }}
```
Comment (tkuzynow) [inline templates/userservice/userservice-deployment.yaml:494]: please test if this is working for other instances like digi where no valeues.global.site24xEnabled is defined

Comment (tkuzynow) [reply templates/userservice/userservice-deployment.yaml:494]: we need to make sure that we're not breaking exisiting pipelines. Probably we'll need to set default value to false here.

**Review by tkuzynow — APPROVED** [review]

**Pull request merged**